dagql/call/id.go

```diff
@@ -243,32 +263,58 @@ func (id *ID) Append(
 		newID.pb.Module = mod.pb
 	}
 
-	for i, arg := range args {
+	for _, arg := range args {
 		if arg.Tainted() {
 			newID.pb.Tainted = true
 		}
-		newID.pb.Args[i] = arg.pb
+		if arg.isSensitive {
+			continue
+		}
+		newID.pb.Args = append(newID.pb.Args, arg.pb)
 	}
 
-	var err error
-	newID.pb.Digest, err = newID.calcDigest()
-	if err != nil {
-		// something has to be deeply wrong if we can't
-		// marshal proto and hash the bytes
-		panic(err)
+	if customDigest != "" {
+		newID.pb.Digest = string(customDigest)
+	} else {
+		var err error
+		newID.pb.Digest, err = newID.calcDigest()
+		if err != nil {
+			// something has to be deeply wrong if we can't
+			// marshal proto and hash the bytes
+			panic(err)
+		}
 	}
 
 	return newID
 }
 
+// Return a new ID that's the same as before except with the given metadata changed.
+// customDigest, if not empty string, will become the ID's digest.
+// tainted sets the ID's tainted flag.
+func (id *ID) WithMetadata(customDigest digest.Digest, tainted bool) *ID {
```

Comment:
Curious what the interplay between this and elective purity is - do selectors even need to be able to have the `Pure` field anymore? Since you can just do `WithMetadata` to taint the instance after the fact.

(and does that mean we could refactor the `CacheVolume` usage as well?)

Reply:
Eyyy :tada:

Can confirm that with that commit, looks like we're back down to "normal" logging again - back down to our regular 50MB (will close https://github.com/dagger/dagger/pull/9428 in preference of this significantly better fix)